### DISCUSSION TITLE
Skipping repeated actions runs on the same git tree

### DISCUSSION THREAD

**Discussion by eli-schwartz** [community/community#64405] [Actions]
### Select Topic Area

Product Feedback

### Body

Often the same git commit is pushed to the same repo multiple times, or an identical commit save for e.g. a tweaked commit message. This is especially likely to happen when merging PRs which are already up to date against the merge target.

In such cases, it's possible to guarantee that the workflow result will be the same if re-run, probably with a setting specified by the workflow file. There are only a couple real inputs:
- the repository tree being acted on
- manually unshallowing the repository checkout and doing something against that information
- non-pinned networked resources downloaded in a step
- workflow code/scripts that check the values of variables such as which branch triggered the action

It is my perception that most workflows don't much care about any of that other than the git tree. It would be very helpful to be able to mark a workflow to indicate that it really, truly, only cares about the git tree and that if those same repository contents have already tested successfully, it doesn't need to rerun the CI and test it again.

This would be hugely beneficial in cases where CI times are quite long.

**Comment by juliandunn** [comment]
Hello from Actions PM here. We are internally testing a feature known as _green trees_ to help with this scenario. In other words, if the tree state is the same as something seen previously, then skip running any work and present the same logs and artifacts back to you. One of the things we still need to figure out is how to make the user experience for this clear.
